File: backend/src/EasyVPN.Domain/Entities/ConnectionTicket.cs
using EasyVPN.Domain.Common.Enums;

namespace EasyVPN.Domain.Entities;

public class ConnectionTicket
{
    public Guid Id { get; set; }
    public Guid ConnectionId { get; set; }
    public User Client { get; set; } = null!;
    public ConnectionTicketStatus Status { get; set; }
    public DateTime CreationTime { get; set; }
    public int Days { get; set; }
    public string PaymentDescription { get; set; } = null!;
    public IEnumerable<string> Images { get; set; } = null!;
}

File: frontend/src/api/responses/ConnectionTicket.ts
import PaymentConnectionInfo from '../common/PaymentConnectionInfo';
import { ConnectionTicketStatus } from '../enums/ConnectionTicketStatus';
import User from './User';

export default interface ConnectionTicket extends PaymentConnectionInfo {
    id: string;
    connectionId: string;
    client: User;
    status: ConnectionTicketStatus;
    creationTime: Date;
}
